server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -123,15 +123,24 @@ public static ByteSizeValue readFrom(StreamInput in) throws IOException {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        TransportVersion tv = out.getTransportVersion();
-        if (tv.onOrAfter(BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_1) && tv.before(REVERT_BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_1)) {
+        if (alwaysUseBytes(out.getTransportVersion())) {
             out.writeZLong(sizeInBytes);
         } else {
             out.writeZLong(Math.divideExact(sizeInBytes, desiredUnit.toBytes(1)));
         }
         desiredUnit.writeTo(out);
     }
 
+    private static boolean alwaysUseBytes(TransportVersion tv) {
+        if (tv.between(BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_1, REVERT_BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_1)) {
+            return true;
+        } else if (tv.isPatchFrom(BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_90)) {
+            return true;
+        } else {
+            return false;
```

Comment:
(Written in this way to maintain visual similarity with the `main` version [here](https://github.com/elastic/elasticsearch/pull/122964/files#diff-fcf5d758aa902536ea547a74f3c668455f1969fbeb3ee7ffbdc8b2b23cb01e0bR141).)